core/query/src/main/java/datawave/core/query/configuration/GenericQueryConfiguration.java

```diff
@@ -29,13 +32,19 @@
  * </p>
  *
  */
-public class GenericQueryConfiguration {
+public class GenericQueryConfiguration implements Serializable {
+    // is this execution expected to be checkpointable (changes how we allocate ranges to scanners)
+    private boolean checkpointable = false;
```

Comment:
is a check for `this instanceof CheckpointableQueryConfiguration` not suitable enough for this?

Reply:
My preference is for instanceOf, it's more reliable than a boolean that could be accidentally set by a non-checkpointable query logic

Reply:
the only downside with removing this is that you wouldn't be able to force the query to run in an uncheckpointable way without the boolean.  not sure if we care about that at all.  @ivakegg?

Reply:
there could definitely be a case for a best effort no checkpoint query

Reply:
So the point of this is that an implementation may support checkpointing, however we may not want to run it in a checkpointable way for latency purposes (think UUID lookup which uses the shard query logic and the shard query configuration)

Reply:
If you update the other class to be `enableCheckpoints` or `checkpointsEnabled` can you do the same here?